### PR TITLE
Refactoring the code and adding services

### PR DESCRIPTION
**Pull request by vijaytembugade**
- Added Services Folder
- Refactor the code 
- Added onScrollLOck Hook
- Refactor the routes

### CODE REVIEW

File: src/Contexts/notesContext/notes-context.js [hunk 1]
```diff
@@ -25,19 +29,24 @@ const NotesProvider = ({ children }) => {
...
     } else {
-      console.log("from else");
       noteDispatch({ type: SET_NOTES, payload: [] });
     }
   }, [isLoggedIn]);
+
+  const notes = removeTrashedNotes(noteState.notes);
+  const bookMarkedNotes = getBookMarkedNotes(notes);
+  const mainNotes = getNoNBookMarkedNotes(notes);
+
```
Comment (kshitijthatte) [inline src/Contexts/notesContext/notes-context.js:45]: You can change the name of `getNoNBookMarkedNotes` to `mainNotes` in utils as well.

**Review by kshitijthatte — APPROVED** [review]
Creating the `services` has made the look code look cleaner also good work with separating all the routes to a different component

File: src/Components/EditNotes/EditNotes.js [hunk 2]
```diff
@@ -1,15 +1,29 @@
...
 import "./EditNotes.css";
 
 const EditNotes = () => {
   const { id } = useParams();
+  const { pathname } = useLocation();
   const {
     noteState: { notes },
   } = useNotes();
 
+  const { noteDetailsDispatch: dispatch } = useNoteDetails();
```
Comment (horsemaker) [inline src/Components/EditNotes/EditNotes.js:15]: Why re-assign the dispatch? 🤔

File: src/Components/LoginForm/LoginForm.js [hunk 3]
```diff
@@ -22,16 +22,18 @@ const LoginForm = () => {
...
-        email: email,
-        password: password,
-      });
-      localStorage.setItem("AUTH_TOKEN", responce.data.encodedToken);
-      localStorage.setItem("AUTH_USER", responce.data.foundUser.email);
-      authDispatch({
-        type: LOGIN_SUCCESS,
-      });
-      navigate("/");
+      const responce = await loginService(email, password);
```
Comment (horsemaker) [inline src/Components/LoginForm/LoginForm.js:25]: typo => `response`

File: src/Components/NoteContainer/NoteContainer.js [hunk 4]
```diff
@@ -28,114 +37,101 @@ const ShowAllNotes = ({ note }) => {
...
-        `/api/notes/${note._id}`,
-        {
-          note: { ...note, inTrash: true },
-        },
-        {
-          headers: { authorization: token },
-        }
-      );
-      noteDispatch({ type: SET_NOTES, payload: [...responce.data.notes] });
+      const responce = await noteMoveToTrashService(note, token);
```
Comment (horsemaker) [inline src/Components/NoteContainer/NoteContainer.js:40]: typo `response`

**Review by horsemaker — APPROVED** [review]
Looks good! 👌

**Pull request merged**